src/site/content/en/blog/usb/index.md

```diff
@@ -0,0 +1,369 @@
+Control transfers are especially nice for small configuration commands as
+they get bus priority and have a well defined structure.
+
+And here's the sketch that has been uploaded to the Arduino board.
+
+```arduino
+// Third-party WebUSB Arduino library
+#include <WebUSB.h>
+
+WebUSB WebUSBSerial(1 /* https:// */, "webusb.github.io/arduino/demos");
+
+#define Serial WebUSBSerial
+
+void setup() {
+  Serial.begin(9600);
+  while (!Serial) {
+    ; // Wait for serial port to connect.
+  }
+  Serial.write("WebUSB FTW!");
+  Serial.flush();
+}
+
+void loop() {
+  // Nothing here for now.
+}
+```
+
+The third-party [WebUSB Arduino library] used in the sample code above does
+basically two things:
+- The device acts as a WebUSB device enabling Chrome to read the [landing page
+  URL].
+- It exposes a WebUSB Serial API that you may use to override the default one.
+
+Let's look at the JavaScript code again. Once we get the `device` picked by the
+user, `device.open` simply runs all platform-specific steps to start a session
+with the USB device. Then, we have to select an available USB Configuration
+with `device.selectConfiguration`. Remember that a Configuration specifies how
+the device is powered, its maximum power consumption and its number of
+interfaces. Talking about interfaces, we also need to request exclusive access
+with `device.claimInterface` since data can only be transferred to an interface
+or associated endpoints when the interface is claimed. Finally calling
+`device.controlTransferOut` is needed to set up the Arduino device with the
+appropriate commands to communicate through the WebUSB Serial API.
+
+From there, `device.transferIn` performs a bulk transfer onto the
+device to inform it that the host is ready to receive bulk data. Then, the
+promise is fulfilled with a `result` object containing a [DataView] `data` that
+has to be parsed appropriately.
+
+For those who are familiar with USB, all of this should look pretty familiar.
+
+### I want more
+
+The WebUSB API lets you interact with the all USB transfer/endpoint types:
+
+- CONTROL transfers, used to send or receive configuration or command
+  parameters to a USB device are handled with `controlTransferIn(setup,
+  length)` and `controlTransferOut(setup, data)`.
+- INTERRUPT transfers, used for a small amount of time sensitive data are
+  handled with the same methods as BULK transfers with
+  `transferIn(endpointNumber, length)` and `transferOut(endpointNumber, data)`.
+- ISOCHRONOUS transfers, used for streams of data like video and sound are
+  handled with `isochronousTransferIn(endpointNumber, packetLengths)` and
+ `isochronousTransferOut(endpointNumber, data, packetLengths)`.
+- BULK transfers, used to transfer a large amount of non-time-sensitive data in
+  a reliable way are handled with `transferIn(endpointNumber, length)` and
+ `transferOut(endpointNumber, data)`.
+
+You may also want to have a look at Mike Tsao's [WebLight project] which
```

Comment:
Have you verified that the project is still up?

Reply:
Yep. I still have some devices as well ;)